lib/Doctrine/ORM/ORMInvalidArgumentException.php

```diff
@@ -104,96 +76,54 @@ static function (array $newEntityWithAssociation): string {
         );
     }
 
-    /**
-     * @param object $entry
-     * @psalm-param array<string, string> $associationMapping
-     *
-     * @return ORMInvalidArgumentException
-     */
-    public static function newEntityFoundThroughRelationship(array $associationMapping, $entry)
+    /** @psalm-param array<string, string> $associationMapping */
+    public static function newEntityFoundThroughRelationship(array $associationMapping, object $entry): ORMInvalidArgumentException
```

Comment:
```suggestion
    public static function newEntityFoundThroughRelationship(array $associationMapping, object $entry): self
```

Can we enforce this via PHPCS, btw?

Reply:
@kukulich hi :wave: ! Would you maybe know about this?

Reply:
I think it's solved by `Squiz.Classes.SelfMemberReference`

Reply:
Hmm, probably not, because the sniff is used in Doctrine CS :(

Reply:
It looks I'll have to write the sniff :)